flang/lib/Lower/Allocatable.cpp

```diff
@@ -424,19 +454,15 @@ class AllocateStmtHelper {
   void lower() {
     visitAllocateOptions();
     lowerAllocateLengthParameters();
-    errorManagement.lower(converter, loc, statExpr, errMsgExpr);
-    // Create a landing block after all allocations so that
-    // we can jump there in case of error.
-    if (errorManagement.hasErrorRecovery())
-      TODO(loc, "error recovery");
-
-    // TODO lower source and mold.
+    errorManager.init(converter, loc, statExpr, errMsgExpr);
     if (sourceExpr || moldExpr)
       TODO(loc, "lower MOLD/SOURCE expr in allocate");
-
+    mlir::Value stat;
+    auto insertPt = builder.saveInsertionPoint();
     for (const auto &allocation :
          std::get<std::list<Fortran::parser::Allocation>>(stmt.t))
-      lowerAllocation(unwrapAllocation(allocation));
+      lowerAllocation(unwrapAllocation(allocation), stat);
```

Comment:
I think the stat mlir::Value could it be propagated through an ErrorManager field from one allocation to the other, that would hide the need for a lot of `stat` argument that may be empty, and it would avoid the `mlir::Value&` pattern that is not very usual.

Reply:
Done